src/main/java/mediator/ConcreteMediator.java

```diff
@@ -0,0 +1,287 @@
+        for (ManagedRadioUnit ru : radioUnits) {
+            if (ru.getRadioUnitName().equals(name)) {
+                switch (rfPorts.size()) {
+                    case 2 -> {
+                        carrier = carrierManagement.createWcdmaCarrier(rfPorts, carrierFrequencies, transmittingPower);
+                        carriers.add(carrier);
+                        ru.setupCarrier(carrier);
+                        return;
+                    }
+                    case 4 -> {
+                        carrier = carrierManagement.createLteCarrier(rfPorts, carrierFrequencies, transmittingPower);
+                        carriers.add(carrier);
+                        ru.setupCarrier(carrier);
+                        return;
+                    }
+                    default -> {
+                        System.out.printf(
+                                "[ERROR] Invalid number of RF ports. Requires 2 or 4, but got %s%n", rfPorts.size());
+                        return;
+                    }
+                }
+            }
+        }
+        System.out.printf("[ERROR] No RU with that name %s is registered with the system%n", name);
+    }
+
+    /**
+     * Creates a carrier that will then be registered with the system. This carrier does not need to be added
+     * to an RU upon creation, but will instead be available to be added to an RU at a later time.
+     * Note: The demo 1 implementation does not make use of this functionality.
+     *
+     * @param rfPorts A list of RF ports that will be used for this carrier.
+     * @param carrierFrequencies The frequency band that will be used with this carrier.
+     * @param transmittingPower The transmitting power for this carrier.
+     */
+    @Override
+    public synchronized void createCarrier(List<RfPort> rfPorts, FrequencyBand carrierFrequencies, Double transmittingPower) {
+        switch (rfPorts.size()) {
+            case 2 -> carriers.add(carrierManagement.createWcdmaCarrier(rfPorts, carrierFrequencies, transmittingPower));
+            case 4 -> carriers.add(carrierManagement.createLteCarrier(rfPorts, carrierFrequencies, transmittingPower));
+            default -> {
+                System.out.printf(
+                        "[ERROR] Invalid number of RF ports. Requires 2 or 4, but got %s%n", rfPorts.size());
+                return;
+            }
+        }
+    }
+
+    /**
+     * Displays the carriers associated with an RU.
+     *
+     * @param name The name of the RU for which the carriers will be displayed.
+     */
+    @Override
+    public void displayCarrierOnRu(String name) {
+        AtomicBoolean flag = new AtomicBoolean(false);
+        radioUnits.forEach(ru -> {
+            if (ru.getRadioUnitName().equals(name)) {
+                flag.set(true);
+                if (ru.getCarriers().size() == 0) {
+                    System.out.printf(
+                            "[ERROR] The RU with the name %s has no carriers associated with it.%n", name);

```

Comment:
I'm not convinced this is necessarily an error condition - maybe we just print the number of carriers (regardless of if there are any there), then list the ones there.
Output might look something like:
```
Number of  carriers: 0
```
or
```
Number of carriers: 1
<carrier details>
```